### PR TITLE
[4-1] refactor timer component

### PR DESCRIPTION
**Pull request by boburwebdev**

### CODE REVIEW

File: exalearn-ui/src/app/services/timer.service.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			seconds = this.speakingTimer % 60;
+
+		const objTime = this.formatTime(minutes, seconds);
+		console.log('working');
+		this.speakingTimer++;
+
+		// if timer is up 5mins, set min/sec to '05/00' and clear interval
+		if (this.speakingTimer >= SPEAKING_DURATION) {
+			mins = '05';
+			secs = '00';
```
Comment (vstrabko) [inline exalearn-ui/src/app/services/timer.service.ts:51]: Could you please move this into const too?

**Review by vstrabko — CHANGES_REQUESTED** [review]

File: exalearn-ui/src/app/services/timer.service.ts (hunk 1 above)
Comment (boburwebdev) [reply exalearn-ui/src/app/services/timer.service.ts:51]: done

**Review by vstrabko — CHANGES_REQUESTED** [review]
One more thing.
Could you please declare timer component into timer module? (Similarly to how it's done with other modules) 
It is obligatory to be able to use it in multiple components with overlapping imports.

File: exalearn-ui/src/app/services/timer.service.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+const SPEAKING_MINS = '05';
+@Injectable({
+	providedIn: 'root'
+})
+export class TimerService {
+	testTotalTimer: number = TEST_DURATION;
+	speakingTimer: number = 0;
+
+	displayTimeLeft(interval: any, mins: string, secs: string) {
+		let minutes = Math.floor(this.testTotalTimer / 60),
```
Comment (vstrabko) [inline exalearn-ui/src/app/services/timer.service.ts]: please, use const here

File: exalearn-ui/src/app/services/timer.service.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+			mins = objTime.mins;
+			secs = objTime.secs;
+		}
+
+		return { mins, secs };
+	}
+
+	// Add zero to mins/secs if they are below 10;
+	formatTime(minutes: number, seconds: number) {
+		let mins = minutes < 10 ? `0${minutes}` : String(minutes);
```
Comment (vstrabko) [inline exalearn-ui/src/app/services/timer.service.ts]: also use const here

**Review by vstrabko — CHANGES_REQUESTED** [review]

File: exalearn-ui/src/app/services/timer.service.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+		return { mins, secs };
+	}
+
+	displayTimePassed(interval: any, mins: string, secs: string) {
+		let minutes = Math.floor(this.speakingTimer / 60),
+			seconds = this.speakingTimer % 60;
+
+		const objTime = this.formatTime(minutes, seconds);
+		console.log('working');
```
Comment (vstrabko) [inline exalearn-ui/src/app/services/timer.service.ts]: also remove console.log()

File: exalearn-ui/src/app/services/timer.service.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+			secs = objTime.secs;
+		}
+
+		return { mins, secs };
+	}
+
+	// Add zero to mins/secs if they are below 10;
+	formatTime(minutes: number, seconds: number) {
+		let mins = minutes < 10 ? `0${minutes}` : String(minutes);
+		let secs = seconds < 10 ? `0${seconds}` : String(seconds);
```
Comment (vstrabko) [inline exalearn-ui/src/app/services/timer.service.ts]: and here

**Review by KotdenGit — APPROVED** [review]

**Review by vstrabko — APPROVED** [review]
Nice use of props to init timer!
But please, show Olga how to init timer by changing passed prop.

**Review by HrebenYury — APPROVED** [review]

File: exalearn-ui/src/app/services/timer.service.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+	}
+
+	displayTimePassed(interval: any, mins: string, secs: string) {
+		const minutes = Math.floor(this.speakingTimer / 60),
+			seconds = this.speakingTimer % 60;
+
+		const objTime = this.formatTime(minutes, seconds);
+		this.speakingTimer++;
+
+		// if timer is up 5mins, set min/sec to '05/00' and clear interval
```
Comment (YanaBr) [inline exalearn-ui/src/app/services/timer.service.ts]: Remove comment about 5 min, we have no this already

**Review by YanaBr — APPROVED** [review]

**Pull request merged**